Pinta/WindowShell.cs

```diff
@@ -33,18 +32,24 @@ public sealed class WindowShell
 {
 	private readonly Gtk.ApplicationWindow app_window;
 	private readonly Adw.HeaderBar? header_bar;
-	private readonly Box shell_layout;
-	private Box? workspace_layout;
-	private Box? main_toolbar;
-
-	public WindowShell (Application app, string name, string title, int width, int height, bool maximize)
+	private readonly Gtk.Box shell_layout;
+	private Gtk.Box? workspace_layout;
+	private Gtk.Box? main_toolbar;
+
+	public WindowShell (
+		Gtk.Application app,
+		string name,
+		string title,
+		int width,
+		int height,
+		bool useHeaderBar,
+		bool maximize)
 	{
 		var app_layout = Adw.ToolbarView.New ();
 
 		// On macOS the global menubar is used, but otherwise use a header bar.
```

Comment:
This comment about the logic for macOS should also be extracted out to the calling code now (`useHeaderBar: PintaCore.System.OperatingSystem != OS.Mac`)

Reply:
Good point. I've now extracted the comment